src/com/google/javascript/jscomp/DisambiguateProperties.java

```diff
@@ -525,6 +531,81 @@ private void handleObjectLit(NodeTraversal t, Node n) {
       }
     }
 
+    private void handleCall(NodeTraversal t, Node n) {
+      Node target = n.getFirstChild();
+      if (target == null || !target.isName()) {
+        return;
+      }
+
+      String renameFunctionName = target.getOriginalQualifiedName();
+      if (renameFunctionName == null ||
+          !compiler.getCodingConvention().isPropertyRenameFunction(renameFunctionName)) {
+        return;
+      }
+
+      if (n.getChildCount() != 3) {
+        compiler.report(JSError.make(
+            n,
+            Warnings.INVALID_RENAME_FUNCTION,
+            renameFunctionName,
+            " Must be called with exactly 2 arguments"));
+        return;
+      }
+
+      if (!n.getSecondChild().isString()) {
+        compiler.report(JSError.make(
+            n,
+            Warnings.INVALID_RENAME_FUNCTION,
+            renameFunctionName,
+            " The first argument must be a string literal."));
+        return;
+      }
+
+      String propName = n.getSecondChild().getString();
+
+      if (propName.split("\\.").length > 1) {
+        compiler.report(JSError.make(
+            n,
+            Warnings.INVALID_RENAME_FUNCTION,
+            renameFunctionName,
+            " The first argument must not be a property path."));
+        return;
+      }
+
+      Node obj = n.getChildAtIndex(2);
+      JSType type = getType(obj);
+      Property prop = getProperty(propName);
+      if (!prop.scheduleRenaming(
+          n.getSecondChild(),
+          processProperty(t, prop, type, null))
+          && propertiesToErrorFor.containsKey(propName)) {
+        String suggestion = "";
+        if (type instanceof JSType) {
```

Comment:
That's a direct copy and paste from the `handleGetProp` function - I didn't even attempt to understand it.